packages/toast/toast-container.js

```diff
@@ -0,0 +1,138 @@
+import { LitElement, css, html } from 'lit';
+import { toaster as c } from '@fabric-ds/component-classes';
+import { repeat } from 'lit/directives/repeat.js';
+
+/**
+ * Toast helper function options
+ * @typedef    {Object}                               ToastOptions
+ * @property   {(number|string)}                      [id]        Custom identifier
+ * @property   {('success'|'error'|'warning'|'info')} [type]      Type of alert
+ * @property   {String}                               [text]      The toast message. Only needed when updating text on existing toast
+ * @property   {(number|string)}                      [duration]  Duration of toast in milliseconds Set to 0 to disable auto-removal
+ * @property   {Boolean}                              [canClose]  Can toast be dismissed?
+ */
+
+export class FabricToastContainer extends LitElement {
+    static styles = css`:host { display: block; }`;
+  
+    static properties = {
+        _toasts: { state: true },
+    };
+
+    constructor() {
+        super();
+        this._toasts = new Map();
+    }
+
+    connectedCallback() {
+        super.connectedCallback();
+        if (this.duration !== Number.POSITIVE_INFINITY) {
+            // regularly check if any toasts have expired
+            this._interval = setInterval(() => {
+                // sort toasts into keep and remove
+                const keep = [];
+                const remove = [];
+                for (const toast of this._toasts) {
+                    if (Date.now() <= toast[1].duration) keep.push(toast);
+                    else remove.push(toast);
+                }
+                // collapse toasts that will be removed
+                const collapseTasks = [];
+                for (const [id] of remove) {
+                    const el = this.renderRoot.querySelector(`#${id}`);
+                    collapseTasks.push(el.collapse());
+                }
+                // once all toasts that should be removed have been collapsed, remove them from the map
+                Promise.all(collapseTasks).then(() => {
+                    if (keep.length != this._toasts.size) this._toasts = new Map(keep);
+                });
+            }, 500);
+        }
```

Comment:
The Fabric Toast Container holds a map of toasts and maintains a timer to regularly check if any of its toasts have expired. 

Any expired toasts are first collapsed visually (animation) before being removed from the set of toasts. Removal from the set of toasts then triggers a re-render which removes the toasts from the dom.